### PR TITLE
feat: simple typeahead search

### PR DESCRIPTION
**Pull request by mvexel**
This adds a simple typeahead search to the layer and style overview. Useful for instances with a lot of layers!

https://github.com/user-attachments/assets/e230b8f9-0c2e-44b7-a095-f2b33f501bb7

### CODE REVIEW

File: public/templates/index.tmpl [hunk 1]
```diff
@@ -13,12 +13,42 @@
...
+          continue;
+        }
+        var details = item.getElementsByClassName('details')[0];
+        var name = details.getElementsByTagName('h3')[0].textContent.toLowerCase();
+        var identifier = details.getElementsByClassName('identifier')[0].textContent.toLowerCase().substring(12); // remove "identifier: "
+        if (name.indexOf(filter) > -1 || identifier.indexOf(filter) > -1) {
+          item.style.display = 'block';
+        } else {
+          item.style.display = 'none';
+        }
```
Comment (akx) [inline public/templates/index.tmpl:32]:
```suggestion
        item.hidden = !(name.indexOf(filter) > -1 || identifier.indexOf(filter) > -1);
```

File: public/templates/index.tmpl [hunk 2]
```diff
@@ -13,12 +13,42 @@
...
+
+    function filter() {
+      var filter = document.getElementById('filter').value.toLowerCase();
+      var items = document.getElementsByClassName('item');
+      for (var i = 0; i < items.length; i++) {
+        var item = items[i];
+        if (!item.getElementsByClassName('details')[0]) {
+          continue;
+        }
+        var details = item.getElementsByClassName('details')[0];
```
Comment (akx) [inline public/templates/index.tmpl:25]:
```suggestion
        var details = item.getElementsByClassName('details')[0];
        if(!details) continue;
```

File: public/templates/index.tmpl [hunk 3]
```diff
@@ -13,12 +13,42 @@
...
+        var name = details.getElementsByTagName('h3')[0].textContent.toLowerCase();
+        var identifier = details.getElementsByClassName('identifier')[0].textContent.toLowerCase().substring(12); // remove "identifier: "
```
Comment (akx) [inline public/templates/index.tmpl:27]:
```suggestion
        var identifier = details.getElementsByClassName('identifier')[0].textContent.toLowerCase().replace(/^identifier: /, '');
```

**Review by akx — COMMENTED** [review]
I'd suggest adding [custom data attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*) to the elements being filtered, so we don't need to traverse the DOM as much :)

**Comment by mvexel** [comment]
@akx haha yes that would be better for sure!
Do you otherwise see this as a useful thing to merge (before I spend more time on it)?

**Comment by acalcutt** [comment]
I think this looks like a nice feature i would merge.

**Comment by mvexel** [comment]
Thanks @acalcutt! I will make the improvements @akx suggested to reduce strain from DOM traversal.

**Comment by mvexel** [comment]
all done and as promised we are *actually* filtering on both name and identifier now :) 
We could extend this with another data attribute to filter on style, layer only with some checkboxes but I propose not overcomplicate things.
Thanks for considering!

**Comment by akx** [comment]
Looks nice and clean with `hidden` and the dataset stuff! Thanks for taking my suggestions into account :)

**Comment by mvexel** [comment]
> Looks nice and clean with `hidden` and the dataset stuff! Thanks for taking my suggestions into account :)

Thanks for the suggestions! I hope this is ready to merge but please do let me know if you have any additional changes you would like to see.

**Comment by acalcutt** [comment]
Just an FYI, I will try to test this soon so it can be merged. I have just been slammed this week at my real job and haven't had much time to test PRs this week.

**Review by acalcutt — APPROVED** [review]
Seems simple and worked fine in testing.

**Comment by acalcutt** [comment]
The one thing I haven't liked about this is how much room the logo + filter take up. one of these days I need to shrink that logo

**Comment by mvexel** [comment]
Have to agree with you. With a little bit of work the filter element could be made smaller. Honestly I think the entire page could benefit from a tighter layout. Let me think about this a bit.

**Pull request merged**